sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md

```diff
@@ -0,0 +1,268 @@
+- [Migration samples](#migration-samples)
+  - [Creating a Container](#creating-a-container)
+  - [Uploading Blobs to a Container](#uploading-blobs-to-a-container)
+  - [Downloading Blobs from a Container](#downloading-blobs-from-a-container)
+  - [Listing Blobs in a Container](#listing-blobs-in-a-container)
+  - [Other](#other)
+- [Additional samples](#additional-samples)
+
+## Migration benefits
+
+To understand why we created our version 12 client libraries, you may refer to the Tech Community blog post, [Announcing the Azure Storage v12 Client Libraries](https://techcommunity.microsoft.com/t5/azure-storage/announcing-the-azure-storage-v12-client-libraries/ba-p/1482394).
+
+In summary, the v12 client libraries for Azure Storage were created in order to address a number of areas of feedback. One of the most important being that Azure services have not been consistent in organization, naming, and API structure. Azure Storage has aligned with a set of common guidelines in order to focus on quick delivery of service features and to optimize the learning curve that comes with using the libraries. They are idiomatic, approachable, and consistent. The version 12 client libraries are also thread safe, offer both synchronous and asynchronous APIs, and have improved performance over previous versions. We have reached feature parity for major scenarios and have updated documentation featuring the v12 [Quickstart](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-dotnet), [Samples](https://docs.microsoft.com/en-us/azure/storage/common/storage-samples-dotnet?toc=/azure/storage/blobs/toc.json), and [Reference](https://docs.microsoft.com/en-us/dotnet/api/azure.storage.blobs?view=azure-dotnet) pages (for .NET Azure Storage Blob Library).
+
+Note: The blog post linked above announces deprecation for previous versions of the library.
+
+## General changes
+
+### Package and namespaces
+
+Package names and the namespaces root for version 12 Azure client libraries follow the pattern `Azure.[Area].[Service]` where the legacy libraries followed the pattern `Microsoft.Azure.[Area].[Service]`.
+
+In this case, to install the legacy v11 package with Nuget:
+```
+dotnet add package Azure.Storage.Blobs
+```
+
+It is now the following for v12:
+```
+dotnet add package Microsoft.Azure.Storage.Blob
+```
+
+### Authentication
+
+#### Managed Identity
+
+Legacy (v11)
+
+v12
+
+#### SAS
+
+There are various SAS tokens that may be generated. Visit our documentation pages to learn how to [Create a User Delegation SAS](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-user-delegation-sas-create-dotnet), [Create a Service SAS](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-service-sas-create-dotnet), or [Create an Account SAS](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-sas-create-dotnet?toc=/azure/storage/blobs/toc.json).
+
+Legacy (v11)
+```c
+
+```
+
+v12
+```c
```

Comment:
In the future maybe we can try to use code snippets from the samples, that way the snippets are frequently ran and checked.

Reply:
I agree that we should in the future! I didn't do that this time because: 
1. It wouldn't have been consistent bc we don't have all the samples
2. The snippets are rather large for some and as opposed to a whole test method I wanted to focus on just the lines that mattered. Maybe not a good approach though.